components/lark-type-check/src/base_inference.rs

```diff
@@ -233,7 +234,14 @@ where
                 }
 
                 if data1.kind != data2.kind {
-                    self.record_error("Mismatched types", cause);
+                    self.record_error(
+                        format!(
+                            "mismatched types ({} vs {})",
```

Comment:
we have to decide on our style of course, but I think we should write use backticks:

```
mismatched types (`{}` vs `{}`)
```